Makefile.one

```diff
@@ -0,0 +1,468 @@
+		-D__BPF_TRACING__ \
+		-D__KERNEL__ \
+		-include $(KERN_SRC_PATH)/include/linux/kconfig.h \
+		-I $(KERN_SRC_PATH)/arch/$(LINUX_ARCH)/include \
+		-I $(KERN_SRC_PATH)/arch/$(LINUX_ARCH)/include/uapi \
+		-I $(KERN_BUILD_PATH)/arch/$(LINUX_ARCH)/include/generated \
+		-I $(KERN_BUILD_PATH)/arch/$(LINUX_ARCH)/include/generated/uapi \
+		-I $(KERN_SRC_PATH)/include \
+		-I $(KERN_BUILD_PATH)/include \
+		-I $(KERN_SRC_PATH)/include/uapi \
+		-I $(KERN_BUILD_PATH)/include/generated \
+		-I $(KERN_BUILD_PATH)/include/generated/uapi \
+		-I$(OUTPUT_DIR)/tracee.bpf \
+		-Wunused \
+		-Wall \
+		-Wno-frame-address \
+		-Wno-unused-value \
+		-Wno-unknown-warning-option \
+		-Wno-pragma-once-outside-header \
+		-Wno-pointer-sign \
+		-Wno-gnu-variable-sized-type-not-at-end \
+		-Wno-deprecated-declarations \
+		-Wno-compare-distinct-pointer-types \
+		-Wno-address-of-packed-member \
+		-fno-stack-protector \
+		-fno-jump-tables \
+		-fno-unwind-tables \
+		-fno-asynchronous-unwind-tables \
+		-xc -O2 -g -emit-llvm \
+		-c $(TRACEE_EBPF_OBJ_SRC) \
+		-o $(@:.o=.ll)
+	$(CMD_LLC) \
+		-march=bpf -mcpu=$(BPF_VCPU) \
+		-filetype=obj \
+		-o $@ \
+		$(@:.o=.ll)
+	$(CMD_RM) $(@:.o=.ll)
+
+.PHONY: clean-bpf-nocore
+clean-bpf-nocore:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee.bpf.$(BPF_NOCORE_TAG).o
+
+#
+# co-re ebpf
+#
+
+TRACEE_EBPF_OBJ_CORE_HEADERS = $(shell find tracee-ebpf -name *.h ! -iregex '.*3rdparty.*')
+
+.PHONY: bpf-core
+bpf-core: $(OUTPUT_DIR)/tracee.bpf.core.o
+
+$(OUTPUT_DIR)/tracee.bpf.core.o: $(OUTPUT_DIR)/libbpf/libbpf.a \
+	$(TRACEE_EBPF_OBJ_SRC) \
+	$(TRACEE_EBPF_OBJ_CORE_HEADERS)
+	$(MAKE) $(OUTPUT_DIR)/tracee.bpf
+	$(CMD_CLANG) \
+		-D__TARGET_ARCH_$(LINUX_ARCH) \
+		-D__BPF_TRACING__ \
+		-DCORE \
+		-I./tracee-ebpf/tracee/co-re/ \
+		-I$(OUTPUT_DIR)/tracee.bpf \
+		-target bpf \
+		-O2 -g \
+		-target bpf -march=bpf -mcpu=$(BPF_VCPU) \
+		-c $(TRACEE_EBPF_OBJ_SRC) \
+		-o $@
+
+.PHONY: clean-bpf-core
+clean-bpf-core:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee.bpf.core.o
+
+#
+# tracee-ebpf
+#
+
+STATIC ?= 0
+GO_TAGS_EBPF = core,ebpf
+CGO_EXT_LDFLAGS_EBPF =
+
+ifeq ($(STATIC), 1)
+    CGO_EXT_LDFLAGS_EBPF += -static
+    GO_TAGS_EBPF := $(GO_TAGS_EBPF),netgo
+endif
+
+CUSTOM_CGO_CFLAGS = "-I$(abspath $(OUTPUT_DIR)/libbpf)"
+CUSTOM_CGO_LDFLAGS = "-lelf -lz $(abspath $(OUTPUT_DIR)/libbpf/libbpf.a)"
+
+GO_ENV_EBPF =
+GO_ENV_EBPF += GOOS=linux
+GO_ENV_EBPF += CC=$(CMD_CLANG)
+GO_ENV_EBPF += GOARCH=$(GO_ARCH)
+GO_ENV_EBPF += CGO_CFLAGS=$(CUSTOM_CGO_CFLAGS)
+GO_ENV_EBPF += CGO_LDFLAGS=$(CUSTOM_CGO_LDFLAGS)
+
+TRACEE_EBPF_SRC=$(shell find ./tracee-ebpf/ -type f -name '*.go' ! -name '*_test.go')
+
+.PHONY: tracee-ebpf
+tracee-ebpf: $(OUTPUT_DIR)/tracee-ebpf
+
+$(OUTPUT_DIR)/tracee-ebpf: $(OUTPUT_DIR)/tracee.bpf.core.o $(TRACEE_EBPF_SRC)
+	$(GO_ENV_EBPF) $(CMD_GO) build \
+		-tags $(GO_TAGS_EBPF) \
+		-ldflags="-w \
+			-extldflags \"$(CGO_EXT_LDFLAGS_EBPF)\" \
+			-X main.version=\"$(VERSION)\" \
+			" \
+		-v -o $@ ./tracee-ebpf
```

Comment:
shouldn't we include `embedded-ebpf.go` from the root as well?

Reply:
Yep, embedded-ebpf.go was created later so it wasn't included. Adding it.

Reply:
wdym when rebuilding the entire project? if you build just this target, does it work? Unless I'm missing something I thing it'll generate a binary without the embeds

Reply:
This target depends on `tracee.bpf.core.o`

`tracee.bpf.core.o` will execute a

`$(MAKE) $(OUTPUT_DIR)/tracee.bpf`

EACH time `libbpf.a` needs rebuild (or tracee ebpf obj src/headers). 

If you change embedded-ebpf.go, then "go build" will be executed again with either:

1. existing contents of the previous "tracee-ebpf" build (which is ok since they haven't been touched)
2. run all dependencies again, since `dist/tracee.ebpf` is always re-created (now) each time you have to build the CO-RE object (or each time you change libbpf, which also needs to rebuild CO-RE object).

Reply:
Maybe I didn't explain the question right: The command in the makefile is essentially go build ./tracee-ebpf . Then how does the go compiler finds the embedding file in the root /

Reply:
I think this answers my question (from `go help build`)

> Build compiles the packages named by the import paths,
along with their dependencies, but it does not install the results.
If the arguments to build are a list of .go files from a single directory,
build treats them as a list of source files specifying a single package.